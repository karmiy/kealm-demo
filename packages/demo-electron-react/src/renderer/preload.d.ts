import { IpcRendererEvent } from 'electron';
import { CHANNEL } from '../common/constants';
import { RenderMessageChannel } from '../main/renderMessageChannel';

declare global {
  interface Window {
    electron: {
      ipcRenderer: {
        sendMessage(channel: CHANNEL, ...args: unknown[]): void;
        invokeMessage(channel: CHANNEL, ...args: unknown[]): Promise<any>;
        on(
          channel: string,
          func: (_event: IpcRendererEvent, ...args: unknown[]) => void
        ): (() => void) | undefined;
        once(
          channel: string,
          func: (_event: IpcRendererEvent, ...args: unknown[]) => void
        ): void;
      };
      messageChannel: {
        get(channel: CHANNEL): RenderMessageChannel;
      };
      darkMode: {
        toggle: () => boolean;
        system: () => void;
      };
      dragger: {
        start: (fileName: string) => void;
      };
      notification: {
        show: (options: { title: string; body: string }) => void;
      };
      progressBar: {
        start: () => void;
      };
    };
  }
}

export {};
