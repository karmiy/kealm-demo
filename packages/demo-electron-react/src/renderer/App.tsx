import {
  MemoryRouter as Router,
  Routes,
  Route,
  useNavigate,
} from 'react-router-dom';
import icon from '../../assets/icon.svg';
import './App.css';
import IPC from './pages/ipc';
import MessageChannel from './pages/messageChannel';
import DarkMode from './pages/darkMode';
import Shortcut from './pages/shortcut';
import DeepLinks from './pages/deepLinks';
import DragFile from './pages/dragFile';
import Notification from './pages/notification';
import NetworkLine from './pages/networkLine';
import ProgressBar from './pages/progressBar';

const Hello = () => {
  const navigate = useNavigate();

  return (
    <div>
      <div className="Hello">
        <img width="200px" alt="icon" src={icon} />
      </div>
      <h1>electron-react-boilerplate</h1>
      <div className="Hello">
        <a
          className="m-10"
          href="https://electron-react-boilerplate.js.org/"
          target="_blank"
          rel="noreferrer"
        >
          <button type="button">
            <span role="img" aria-label="books">
              📚
            </span>
            Read Docs
          </button>
        </a>
        <a
          className="m-10"
          href="https://github.com/sponsors/electron-react-boilerplate"
          target="_blank"
          rel="noreferrer"
        >
          <button type="button">
            <span role="img" aria-label="books">
              🙏
            </span>
            Donate
          </button>
        </a>
      </div>
      <div className="Hello">
        <button className="m-10" type="button" onClick={() => navigate('ipc')}>
          <span role="img" aria-label="books" aria-hidden="true">
            🎉
          </span>
          IPC Page
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('message-channel')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🍕
          </span>
          Message Page
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('dark-mode')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🎏
          </span>
          Dark Mode
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('shortcut')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            🎄
          </span>
          Shortcut
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('deep-links')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📙
          </span>
          Deep Links
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('drag-file')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📒
          </span>
          Drag File
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('notification')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            ✉️
          </span>
          Notification
        </button>
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('network-line')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            📡
          </span>
          NetworkLine
        </button>
      </div>
      <div className="Hello">
        <button
          className="m-10"
          type="button"
          onClick={() => navigate('progress-bar')}
        >
          <span role="img" aria-label="books" aria-hidden="true">
            💿
          </span>
          ProgressBar
        </button>
      </div>
    </div>
  );
};

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Hello />} />
        <Route path="/ipc" element={<IPC />} />
        <Route path="/message-channel" element={<MessageChannel />} />
        <Route path="/dark-mode" element={<DarkMode />} />
        <Route path="/shortcut" element={<Shortcut />} />
        <Route path="/deep-links" element={<DeepLinks />} />
        <Route path="/drag-file" element={<DragFile />} />
        <Route path="/notification" element={<Notification />} />
        <Route path="/network-line" element={<NetworkLine />} />
        <Route path="/progress-bar" element={<ProgressBar />} />
      </Routes>
    </Router>
  );
}
