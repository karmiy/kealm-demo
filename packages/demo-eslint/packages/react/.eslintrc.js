module.exports = {
    extends: ['../../eslint-config/eslintrc.react.js', '../../eslint-config/eslintrc.javascript.js'].map(require.resolve),
    // extends: ['../../eslint-config/eslintrc.react.js'],

    // rules: { 'react/jsx-indent': [2, 4] },

    /** 该层级的文件不再使用该层父级以上的 eslintrc 配置 */
    root: true,
};
