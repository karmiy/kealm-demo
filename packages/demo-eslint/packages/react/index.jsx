import React, { useState, Component, useEffect } from 'react';
import PropTypes from 'prop-types';

function Button(props) {
    const [count, setCount] = useState(0);

    useEffect(() => {
        window.addEventListener('click', () => setCount(count + 1));
    }, [count]);

    const { id, children } = props;

    return (
        <div>
            {count}
            {id}
            <p />
            <button
                type='button'
                className='km-button'
                onClick={() => {
                    setCount(count + 1);
                }}
            >
                {children}

            </button>
        </div>
    );
}

Button.propTypes = { id: PropTypes.number };
Button.defaultProps = { id: 1 };

class Wrap extends Component {
    id = 10;

    /** 123 */
    onClick() {
        const { name } = this.props;
        console.log(this.id);
        console.log(name);
    }

    say() {
        console.log(1, this);
    }

    render() {
        return (
            <div>
                wrap:
                {this.id}
            </div>
        );
    }
}

Wrap.propTypes = { name: PropTypes.string.isRequired };

const obj = {
    id: 1,
    name: 'k',
};

console.log(obj);

export {
    Wrap,
    Button,
};