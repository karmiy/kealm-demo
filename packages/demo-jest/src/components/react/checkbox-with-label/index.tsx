import React, { useState, useCallback } from 'react';
import { ICheckboxWithLabelProps } from './interface';

const CheckboxWithLabel: React.FC<ICheckboxWithLabelProps> = props => {
    const [isChecked, setIsChecked] = useState(false);
    const { labelOn, labelOff } = props;

    const onChange: React.ChangeEventHandler<HTMLInputElement> = useCallback(() => setIsChecked(checked => !checked), []);
    
    return (
        <label>
            <input
                type='checkbox'
                checked={isChecked}
                onChange={onChange}
            />
            {isChecked ? labelOn : labelOff}
        </label>
    )
};

export default CheckboxWithLabel;

