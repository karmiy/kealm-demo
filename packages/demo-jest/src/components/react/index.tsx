export { default as Link } from './link';
export { default as CheckboxWithLabel } from './checkbox-with-label';