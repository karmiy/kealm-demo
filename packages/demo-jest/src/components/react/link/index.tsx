import React, { useState, useCallback } from 'react';
import { ILinkProps } from './interface';

enum STATUS {
    HOVERED = 'hovered',
    NORMAL = 'normal',
}

const Link: React.FC<ILinkProps> = props => {
    const [className, setClassName] = useState(STATUS.NORMAL);
    const { page = '#', children } = props;

    const onMouseEnter = useCallback(() => setClassName(STATUS.HOVERED), []);

    const onMouseLeave = useCallback(() => setClassName(STATUS.NORMAL), []);

    return (
        <a
            className={className}
            href={page}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
        >
            {children}
        </a>
    )
};

export default Link;