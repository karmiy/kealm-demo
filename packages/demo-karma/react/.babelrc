{
    "presets": [
      "@babel/preset-typescript",
      "@babel/preset-react",
      ["@babel/preset-env", { "targets": { "node": "current" } }]
    ],
    "plugins": [
      "babel-plugin-inline-import-data-uri",
      "@babel/plugin-transform-member-expression-literals",
      "@babel/plugin-transform-object-assign",
      "@babel/plugin-transform-property-literals",
      ["@babel/plugin-transform-runtime", { "helpers": false }],
      "@babel/plugin-transform-spread",
      "@babel/plugin-transform-template-literals",
      "@babel/plugin-proposal-export-default-from",
      "@babel/plugin-proposal-export-namespace-from",
      "@babel/plugin-proposal-object-rest-spread",
      ["@babel/plugin-proposal-decorators", { "legacy": true }],
      "@babel/plugin-proposal-class-properties",
      "@babel/plugin-syntax-dynamic-import"
    ]
}