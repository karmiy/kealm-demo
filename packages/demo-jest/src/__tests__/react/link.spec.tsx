import React from 'react';
/* 建议将任何呈现和触发组件更新的代码封装到 act() 调用中 */
// https://reactjs.org/blog/2019/02/06/react-v16.8.0.html#testing-hooks
import renderer, { act } from 'react-test-renderer';

/* https://testing-library.com/docs/dom-testing-library */
import { render, fireEvent } from '@testing-library/react';

/* https://enzymejs.github.io/enzyme/ */
/**
 * Enzyme.render 生成 HTML 结构，可用于快照
 * Enzyme.shallow 生成 react tree，只渲染当前组件
 * Enzyme.mount 生成 react tree，渲染当前组件包括子组件
 */
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { Link, CheckboxWithLabel } from '@components/react';


describe('Link', () => {
    /* 快照测试 */
    it('test snapshot that Link changes the class when hovered', async () => {
        const component = renderer.create(
            <Link page='https://github.com/karmiy/demo-jest'>demo-jest</Link>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => tree?.props.onMouseEnter()); // 需要用 act 包装带副作用的操作，否则 react hooks 的 setXXX 操作会警告
        tree = component.toJSON(); // 触发 onMouseEnter 事件后重新创建 render tree
        expect(tree).toMatchSnapshot();

        act(() => tree?.props.onMouseLeave());
        tree = component.toJSON();  // 触发 onMouseLeave 事件后重新创建 render tree
        expect(tree).toMatchSnapshot();
    });

    /* 测试 react-testing-library */
    it('test with react-testing-library that CheckboxWithLabel changes the text after click', () => {
        const { queryByLabelText, getByLabelText } = render(<CheckboxWithLabel labelOn='On' labelOff='Off' />);

        expect(queryByLabelText(/off/i)).toBeTruthy();
        fireEvent.click(getByLabelText(/off/i));
        expect(queryByLabelText(/on/i)).toBeTruthy();
    });

    /* 测试 enzyme */
    it('test with enzyme that CheckboxWithLabel changes the text after click', () => {
        const checkbox = shallow(<CheckboxWithLabel labelOn='On' labelOff='Off' />);

        expect(checkbox.text()).toEqual('Off');
        checkbox.find('input').simulate('change'); // 触发 change 事件
        expect(checkbox.text()).toEqual('On');
    });
});
