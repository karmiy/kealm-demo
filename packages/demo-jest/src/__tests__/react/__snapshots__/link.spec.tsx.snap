// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link test snapshot that Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://github.com/karmiy/demo-jest"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  demo-jest
</a>
`;

exports[`Link test snapshot that Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://github.com/karmiy/demo-jest"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  demo-jest
</a>
`;

exports[`Link test snapshot that Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://github.com/karmiy/demo-jest"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  demo-jest
</a>
`;
